assets/app/view/game/part/town_location.rb

```diff
@@ -0,0 +1,250 @@
+                                                  end
+        end
+
+        def min_edge(edges)
+          @@min_edge ||= {}
+          @@min_edge[edges] ||= edges.min
+        end
+
+        # Returns a symbol of :straight, :gentle or :sharp based on the absolute
+        # difference of the normalized edges
+        def town_track_type(edges)
+          @@town_track_type ||= {}
+          @@town_track_type[edges] ||=
+            begin
+              edge_a, edge_b = edges
+              [nil, :sharp, :gentle, :straight][(edge_a - edge_b).abs]
+            end
+        end
+
+        # Returns a symbol of :left, :right, or :straight based on the direction
+        # the track curves starting from edge_a
+        def town_track_direction(edges)
+          @@town_track_direction ||= {}
+          @@town_track_direction[edges] ||=
+            begin
+              edge_a, edge_b = edges
+
+              if (edge_a - edge_b).abs == 3
+                :straight
+              elsif edge_a > edge_b
+                :right
+              elsif edge_a < edge_b
+                :left
+              end
+            end
+        end
+
+        # Returns an array of options for track positions
+        def town_track_location(tile, town, edges)
+          edge_a, = edges
+
+          loc = if town.exits.size == 2 && tile.stops.one?
+                  SINGLE_STOP_TWO_EXIT_REGIONS[town_track_type(edges)][min_edge(edges)]
+                elsif town.exits.size == 2
+                  MULTIPLE_STOP_TWO_EXIT_REGIONS[town_track_type(edges)][edge_a % 6]
+                elsif edge_a
+                  EDGE_TOWN_REGIONS[edge_a]
+                else
+                  CENTER
+                end
+          [loc]
+        end
+
+        # Returns an array of rotation options for the town rectangle that
+        # corresponds to the positions given from track_location
+        def town_rotation_angles(tile, town, edges)
+          edge_a, = edges
+
+          if town.exits.size == 2 && tile.stops.one?
+            case town_track_type(edges)
+            when :straight
+              [min_edge(edges) * 60]
+            when :sharp
+              [(min_edge(edges) + 2) * 60]
+            when :gentle
+              [(min_edge(edges) * 60) - 30]
+            end
+          elsif town.exits.size == 2
+            tilt = RECTANGLE_TILT[town_track_type(edges)] || 0
+
+            delta = case town_track_direction(edges)
+                    when :straight
+                      0
+                    when :left
+                      -tilt
+                    when :right
+                      tilt
+                    end
+
+            [(edge_a * 60) + delta]
+          elsif edge_a
+            [edge_a * 60]
+          else
+            # This town is in the center. Find the orientation of the town
+            # by looking at the first path connected to it
+            path = town.paths.first
+            if path.edges.any?
```

Comment:
if (edge = path.exits&.first)
[edge * 60]

Reply:
Got it